django_project/realtime/tasks/earthquake.py

```diff
@@ -251,30 +235,69 @@ def generate_earthquake_report(event):
     event.refresh_from_db()
     impact_layer_uri = event.impact_file_path
 
-    result = AsyncResult(id=event.analysis_task_id, app=headless_app)
+    analysis_result = json.loads(event.analysis_task_result)
 
-    layer_order = []
+    source_dict = dict(analysis_result['output'])
+    source_dict.update({
+        'hazard': event.hazard_path
+    })
+
+    layer_order = substitute_layer_order(
+        EARTHQUAKE_LAYER_ORDER, source_dict)
+
+    tasks_chain = chain(
+        # Generate report
+        generate_report.s(
+            impact_layer_uri, EARTHQUAKE_REPORT_TEMPLATE, layer_order
+        ).set(queue=generate_report.queue),
+
+        # Handle report
+        handle_report.s(
+            event.id
+        ).set(queue=handle_report.queue)
+    )
+
+    @app.task
+    def _handle_error(req, exc, traceback):
+        """Update task status as Failure."""
+        Earthquake.objects.filter(id=event.id).update(
+            report_task_status='FAILURE')
+
+    async_result = tasks_chain.apply_async(link_error=_handle_error.s())
 
-    for layer in EARTHQUAKE_LAYER_ORDER:
```

Comment:
Refactor into utils.
I think you can use and improve it for other hazards @ismailsunni